src/seedu/addressbook/commands/Command.java

```diff
@@ -63,6 +63,14 @@ protected ReadOnlyPerson getTargetPerson() throws IndexOutOfBoundsException {
         return relevantPersons.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
     }
 
+    /**
+     * Returns true for command types that mutate the data    
+     * To be overriden by sub classes
```

Comment:
As mentioned in class, having a default implementation for the `isMutating` method helps to reduce the amount of code, but brings ambiguity to the codes (since having `false` or `true` as the default return value is rather arbitrary). Making the method `abstract` forces all child classes to declare whether it is a mutating command or not makes the codes more intuitive and less arbitrary.